libp2p/utility.nim

```diff
@@ -0,0 +1,33 @@
+## Nim-LibP2P
+## Copyright (c) 2020 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import nimcrypto/utils
+
+const
+  ShortDumpMax = 24
+
+func shortHexDump*(item: openarray[byte]): string =
```

Comment:
in the other projects (nbc), we've called this function `shortLog`

Reply:
Cool! I guess I can rename it, not sure if behaves the same tho.

Reply:
kind of - ie we use it for cutting down digests and public keys etc in size

Reply:
Current implementation uses 5 allocations:

1. item[0 ..< split]
2. item[0 ..< split].toHex()
3. item[^split .. item.high]
4. item[^split ..item.high].toHex()
5. aslice & "..." & bslice

But it can use only 3 allocations if you will use `toOpenArray`.

1. item.toOpenArray(0, split - 1).toHex()
2. item.toOpenArray(split - 1, item.high).toHex()
3. aslice & "..." & bslice

Or even 2 if you optimize result concatenation.

Reply:
@cheatfate super cool thanks, I'm having issues with the string version tho:
`invalid type: 'openArray[char]' for let` I will try to figure why

Reply:
Btw I think nim concat `&` does a mess of allocations too.
But well, it's pooled gc mem.. anyway I will fix it